File: runtime/jvm/src/lox/DeferredCall.java
package lox;

/**
 * Captures a callable and its arguments for deferred execution.
 * Represents a single `defer` statement from Lox++.
 */
public class DeferredCall {
    public final Object callable;     // Closure or native function
    public final Object[] args;       // Argument values

    public DeferredCall(Object callable, Object[] args) {
        this.callable = callable;
        this.args = args;
    }
}

File: runtime/clr/src/DeferredCall.cs
namespace Lox;

/// <summary>
/// Captures a callable and its arguments for deferred execution.
/// Represents a single `defer` statement from Lox++.
/// </summary>
public class DeferredCall
{
    public readonly object? Callable;
    public readonly object?[] Args;

    public DeferredCall(object? callable, object?[] args)
    {
        Callable = callable;
        Args = args;
    }
}
